### PR TITLE
Support QuotaUser/Project and IAMAuthToken

### PR DESCRIPTION
**Pull request by shinfan**
- Attach fields into headers in both HTTP and gRPC
- Refactor the gRPC token source code for better readability
- Move HTTP transport code into internal/http.go

### CODE REVIEW

**Comment by shinfan** [comment]
@julianshaoliu FYI

File: go/sgauth/internal/const.go [hunk 1]
```diff
@@ -14,21 +14,11 @@
...
-
-// RequireTransportSecurity indicates whether the credentials requires transport security.
-func (key GrpcApiKey) RequireTransportSecurity() bool {
-	return true
-}
+// Keys for HTTP google headers
+const (
+	headerAuth = "authorization"
+	headerApiKey = "X-Goog-Api-Key"
+	headerQuotaUser = "X-Goog-QuotaUser"
```
Comment (wora) [inline go/sgauth/internal/const.go]: Quota-User

File: go/sgauth/settings.go [hunk 2]
```diff
@@ -35,11 +35,11 @@ type Settings struct {
...
 	QuotaProject string
-
+	// IAM context attributes
+	IAMAuthToken string
 	// End-user OAuth Flow handler that redirects the user to the given URL
 	// and returns the token.
 	OAuthFlowHandler func(url string) (token string, err error)
-
 	// The state string used for 3LO session verification.
 	State string
```
Comment (wora) [inline go/sgauth/settings.go]: Document most features as Unimplemented even you write the code. Many features don't exist yet.

File: go/sgauth/internal/const.go (hunk 1 above)
Comment (shinfan) [reply go/sgauth/internal/const.go]: go/system-parameters documented as "X-Goog-QuotaUser", I will correct the documentation.

Comment (shinfan) [reply go/sgauth/internal/const.go]: The latest version is already corrected to Quota-User. It is just not released yet.

**Comment by shinfan** [comment]
@wora PTAL

**Pull request merged**